llpc/context/llpcCompiler.cpp

```diff
@@ -1561,18 +1571,24 @@ Result Compiler::BuildGraphicsPipeline(const GraphicsPipelineBuildInfo *pipeline
 // @param [out] pipelineElf : Output Elf package
 Result Compiler::buildComputePipelineInternal(ComputeContext *computeContext,
                                               const ComputePipelineBuildInfo *pipelineInfo, bool buildingRelocatableElf,
-                                              ElfPackage *pipelineElf) {
+                                              ElfPackage *pipelineElf, ComputePipelineBuildOut *pipelineOut) {
   Context *context = acquireContext();
   context->attachPipelineContext(computeContext);
 
   std::vector<const PipelineShaderInfo *> shadersInfo = {
       nullptr, nullptr, nullptr, nullptr, nullptr, &pipelineInfo->cs,
   };
   Result result;
-  if (buildingRelocatableElf)
-    result = buildPipelineWithRelocatableElf(context, shadersInfo, pipelineElf);
-  else
+  if (buildingRelocatableElf) {
+    StageCacheInfo stageCacheAccesses[ShaderStageCount] = {};
+    result = buildPipelineWithRelocatableElf(context, shadersInfo, pipelineElf, stageCacheAccesses);
+    assert(llvm::all_of(
+        std::array<StageCacheInfo>(stageCacheAccesses, stageCacheAccesses[ShaderStageCompute]),
```

Comment:
Why not define `stageCacheAccesses` as `std::array`? I'd also think that `any_of` should work with a plain C array of static size, no?

Reply:
The problem is that he wants to leave out the element referring to the compute stage.  I don't think this is worth asserting. The program will still work fine even if the assert fails.  I should be up to `buildPipelineWithRelocatableElf` to make sure it did not do anything unexpected.  It is not the responsibility of each of the users of the function.